lib/auth/okta/auth.go

```diff
@@ -130,29 +125,53 @@ var (
 //
 // If any resource is not requestable, [OktaResourceNotRequestableError] will be returned. Any
 // other error can be returned.
-func CheckResourcesRequestable(ctx context.Context, ids []types.ResourceID, ap AccessPoint) error {
-	bidirectionalSyncEnabled, err := BidirectionalSyncEnabled(ctx, ap.Plugins)
+func CheckResourcesRequestable(ctx context.Context, ids []types.ResourceID, auth AuthServer) error {
+	if len(ids) == 0 {
+		return nil
+	}
+
+	bidirectionalSyncEnabled, err := BidirectionalSyncEnabled(ctx, auth)
 	if err != nil {
 		return trace.Wrap(err, "getting bidirectional sync")
 	}
 	if bidirectionalSyncEnabled {
 		return nil
 	}
 
-	var denied []types.ResourceID
-	for app, err := range ap.UnifiedResourceCache.AppServers(ctx, services.UnifiedResourcesIterateParams{}) {
+	hasApps := slices.ContainsFunc(ids, func(id types.ResourceID) bool {
+		return id.Kind == types.KindAppServer || id.Kind == types.KindApp
+	})
+	var oktaAppServers []types.AppServer
+	if hasApps {
+		oktaAppServers, err = auth.GetApplicationServers(ctx, apidefaults.Namespace)
```

Comment:
Instead of listing all application servers before performing any operation, would it be possible to call:
```
GetUnifiedResourcesByIDs(ctx, id)
```

for the `types.KindAppServer, types.KindApp:` case 

There is no need to load all app servers if the resourcesIDs are known.

Reply:
It can't be done. Added a comment here https://github.com/gravitational/teleport/compare/b66b164e9f3d95f4e8180062f8ee87124f970ebe..e4198668661ab5d886616f2d5d817731830bd1d7